java/src/main/java/dojo/supermarket/model/Offer.java

```diff
@@ -0,0 +1,18 @@
+package dojo.supermarket.model;
+
+public class Offer {
+
+    SpecialOfferType offerType;
+    private final Product product;
+    double argument;
```

Comment:
use private final for non-changing fields
```suggestion
    private final SpecialOfferType offerType;
    private final Product product;
    private final double argument;
```